tests/unit_tests/data_steward/retraction/retract_deactivated_pids_test.py

```diff
@@ -0,0 +1,226 @@
+# Python imports
+import unittest
+import re
+
+# Third party imports
+import mock
+from mock import patch
+import pandas as pd
+from pandas.util.testing import assert_frame_equal
+
+# Project imports
+from retraction import retract_deactivated_pids
+from retraction.retract_utils import DEID_REGEX
+from utils import bq
+from constants import bq_utils as bq_consts
+from constants.cdr_cleaner import clean_cdr as clean_consts
+from sandbox import get_sandbox_dataset_id
+
+
+class RetractDataBqTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.project_id = 'fake_project_id'
+        self.ticket_number = 'DC_fake'
+        self.pids_project_id = 'fake_pids_project_id'
+        self.pids_dataset_id = 'fake_pids_dataset_id'
+        self.pids_table = 'fake_pids_table'
+        self.pids_dataset_list = [
+            'fake_dataset_1', 'fake_dataset_2', 'fake_dataset_3'
+        ]
+        self.pids_table_list = ['fake_table_1', 'fake_table_2', 'fake_table_3']
+
+        mock_bq_client_patcher = patch(
+            'retraction.retract_deactivated_pids.get_client')
+        self.mock_bq_client = mock_bq_client_patcher.start()
+        self.addCleanup(mock_bq_client_patcher.stop)
+
+    def test_get_pids_table_info(self):
+        for dataset in self.pids_dataset_list:
+            result_df = self.mock_bq_client.query(
+                retract_deactivated_pids.TABLE_INFORMATION_SCHEMA.render(
+                    project=self.project_id, dataset=dataset)).to_dataframe()
+            expected_pids_tables = []
+            for row in result_df.iterrows():
+                column = getattr(row, 'column_name')
+                table = getattr(row, 'table_name')
+                if 'person_id' in column:
+                    expected_pids_tables.append(table)
+            expected_pids_tables_info_df = result_df[
+                result_df['table_name'].isin(expected_pids_tables)]
+
+            returned_result_df = retract_deactivated_pids.get_pids_table_info(
+                self.project_id, dataset, self.mock_bq_client)
+            self.assertEqual(expected_pids_tables_info_df, returned_result_df)
```

Comment:
This unit test is only checking that a mocked object equals a mocked object.  So it will always pass.  As it is, it never gets inside the loop at line 50, so none of that ever gets checked.  This test should be rewritten and simplified.

This patch is a unit test that can be incorporated.  It only mocks the return value of the `client.query().to_dataframe()` call and runs the rest of the function.  It sets the expected return value based on the given parameters and asserts a comparison.  It does not rewrite the internals of the function it is testing to generate the expected result.  That test pattern will test that you are getting the same result in both places, not whether the result created by your code is actually correct.

The provided test fails because of the error on line 177 in `retract_deactivated_pids.py`.
[patch.txt](https://github.com/all-of-us/curation/files/4738011/patch.txt)